libvast/vast/system/spawn_source.hpp

```diff
@@ -14,50 +14,59 @@
 #pragma once
 
 #include "vast/aliases.hpp"
+#include "vast/config.hpp"
+#include "vast/defaults.hpp"
+#include "vast/detail/make_io_stream.hpp"
+#include "vast/detail/unbox_var.hpp"
 #include "vast/fwd.hpp"
+#include "vast/logger.hpp"
+#include "vast/system/node.hpp"
+#include "vast/system/source.hpp"
+#include "vast/system/spawn_arguments.hpp"
 
 namespace vast::system {
 
-/// Tries to spawn a new SOURCE for the CSV format.
+/// Tries to spawn a new SOURCE for the specified format.
+/// @tparam Reader the format-specific reader
 /// @param self Points to the parent actor.
 /// @param args Configures the new actor.
 /// @returns a handle to the spawned actor on success, an error otherwise
-maybe_actor spawn_csv_source(node_actor* self, spawn_arguments& args);
-
-/// Tries to spawn a new SOURCE for the JSON format.
-/// @param self Points to the parent actor.
-/// @param args Configures the new actor.
-/// @returns a handle to the spawned actor on success, an error otherwise
-maybe_actor spawn_json_source(node_actor* self, spawn_arguments& args);
-
-/// Tries to spawn a new SOURCE for the PCAP format.
-/// @param self Points to the parent actor.
-/// @param args Configures the new actor.
-/// @returns a handle to the spawned actor on success, an error otherwise
-maybe_actor spawn_pcap_source(node_actor* self, spawn_arguments& args);
-
-/// Tries to spawn a new SOURCE for the Suricata format.
-/// @param self Points to the parent actor.
-/// @param args Configures the new actor.
-/// @returns a handle to the spawned actor on success, an error otherwise
-maybe_actor spawn_suricata_source(node_actor* self, spawn_arguments& args);
-
-/// Tries to spawn a new SOURCE for the Syslog format.
-/// @param self Points to the parent actor.
-/// @param args Configures the new actor.
-/// @returns a handle to the spawned actor on success, an error otherwise
-maybe_actor spawn_syslog_source(node_actor* self, spawn_arguments& args);
-
-/// Tries to spawn a new SOURCE for generating random data.
-/// @param self Points to the parent actor.
-/// @param args Configures the new actor.
-/// @returns a handle to the spawned actor on success, an error otherwise
-maybe_actor spawn_test_source(node_actor* self, spawn_arguments& args);
-
-/// Tries to spawn a new SOURCE for the Zeek format.
-/// @param self Points to the parent actor.
-/// @param args Configures the new actor.
-/// @returns a handle to the spawned actor on success, an error otherwise
-maybe_actor spawn_zeek_source(node_actor* self, spawn_arguments& args);
+template <class Reader, class Defaults = typename Reader::defaults>
+maybe_actor spawn_source(node_actor* self, spawn_arguments& args) {
+  VAST_TRACE(VAST_ARG("node", self), VAST_ARG(args));
+  auto& st = self->state;
+  auto& options = args.inv.options;
+  // Bail out early for bogus invocations.
+  if (caf::get_or(options, "system.node", false))
+    return make_error(ec::parse_error, "cannot start a local node");
```

Comment:
Can we make the error message more meaningful, i.e., *why* can't we start a local node?

Reply:
I fully agree. We use the same error message everywhere we bail out when `system.node` is set, so I just copied it over here.

Reply:
Do you have a suggestion for a better replacement?

Reply:
The error sounds like an invalid combination of command line parameters that are not working, so I'd say something along the lines of "invalid command line: 'spawn' in local node is a no-op".

Reply:
I've created a story for replacing this everywhere in both VAST and VAST Pro.

Reply:
I think it makes sense to change the messages that are introduced in this pr to the proposed wording directly. A new story for doing this everywhere is nice, but it won't get prioritized for a while.